src/test/shell/bazel/starlark_repository_test.sh

```diff
@@ -151,24 +137,19 @@ genrule(
 EOF
   cd ${WORKSPACE_DIR}
   cat >test.bzl <<EOF
+load("@bazel_tools//tools/build_defs/repo:local.bzl", "local_repository")
 def macro(path):
   print('blah')
-  native.local_repository(name='endangered', path='$repo2')
```

Comment:
Do we keep test coverage for `native.local_repository` until we fully remove it?

Reply:
hmm, I'm not sure how to do that without huge changes to the test setup. I'm willing to say "we're not going to allow any more changes to the native repo rules going forward" and declare their behavior stable enough to not need to keep the tests. (I realize that sounds a bit "famous last word"-ish...) WDYT?

Reply:
Sgtm as well, but we should add a comment to the native implementation about it being frozen.

Reply:
done.

Reply:
also I plan to add an incompatible flag to disable the native impls in a follow-up, just FYI